src/main/java/frc/robot/util/DashboardFactory.java

```diff
@@ -0,0 +1,47 @@
+package frc.robot.util;
+
+import java.util.function.BooleanSupplier;
+
+import edu.wpi.first.cameraserver.CameraServer;
+import edu.wpi.first.cscore.VideoCamera;
+import edu.wpi.first.util.PixelFormat;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+public class DashboardFactory {
```

Comment:
this isn't quite what a factory pattern is but I understand what you were going for. Lets talk about this more in person